src/components/charts/BarChart.tsx

```diff
@@ -0,0 +1,419 @@
+  const [startTime, setStartTime] = useState<string>(null);
+  const [startDate, setStartDate] = useState<string>(null);
+  const [startDateTime, setStartDateTime] = useState<string>("");
+  const [channel, setChannel] = useState<string>("all");
+  const [showFilterPopup, setShowFilterPopup] = useState<boolean>(false);
+  const [submitCounter, setSubmitCounter] = useState<number>(0);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getSessionData();
+    // getHardCodedSessionData();
+  }, [submitCounter]);
+
+  useEffect(() => {
+    const labels: string[] = [];
+    const azurePrimary: string[] = [];
+    const azureSecondary: string[] = [];
+
+    sessionData.forEach((item) => {
+      labels.push(item.timestamp);
+      azurePrimary.push(parseInt(item.azureDCSessionsCount));
+      azureSecondary.push(parseInt(item.qtsDCSessionsCount));
+    });
+
+    setXAxisLabels([...labels]);
+    setAzurePrimaryData([...azurePrimary]);
+    setAzureSecondaryData([...azureSecondary]);
+  }, [sessionData]);
+
+  useEffect(() => {
+    const chartData: ChartData = {
+      labels: xAxisLabels,
+      datasets: [
+        {
+          label: "Azure Primary",
+          data: azurePrimaryData,
+          backgroundColor: "#757575",
+        },
+      ],
+    };
+    setPrimaryData({ ...chartData });
+  }, [xAxisLabels, azurePrimaryData]);
+
+  useEffect(() => {
+    const chartData: ChartData = {
+      labels: xAxisLabels,
+      datasets: [
+        {
+          label: "Azure Secondary",
+          data: azureSecondaryData,
+          backgroundColor: "#BABABA",
+        },
+      ],
+    };
+    setSecondaryData({ ...chartData });
+  }, [xAxisLabels, azureSecondaryData]);
+
+  const getSessionData = async () => {
+    const params = {
+      period: duration,
+      starttime: startDateTime,
+      channel: channel,
+    };
+    const data = await fetchData(URL_SESSIONS, params);
+    setIsLoading(false);
+    setSessionData(data || []);
+  };
+
+  // const getHardCodedSessionData = () => {
+  //   setIsLoading(false);
+  //   setSessionData(sessionsJSON);
+  // };
+
+  const changeStartTime = (value: string) => {
+    const formattedTime = formatTime(value);
+    setStartTime(value);
+    let defaultDate: string = "";
+
+    defaultDate = !startDate
+      ? `${new Date().toISOString().split("T")[0]}`
+      : formatDate(startDate, DATE_FORMAT_2);
+
+    setStartDateTime(`${defaultDate}T${formattedTime}`);
+  };
+
+  const changeStartDate = (value: string) => {
+    setStartDate(value);
+    const selectedDate = new Date(value);
+    let defaultTime: string = "";
+    defaultTime = !startTime ? formatTime(new Date()) : formatTime(startTime);
+    setStartDateTime(
+      `${formatDate(selectedDate, DATE_FORMAT_2)}T${defaultTime}`,
+    );
+  };
+
+  const getChartConfig = (type: string) => {
+    const chartOptions = JSON.parse(JSON.stringify(BAR_CHART_OPTIONS));
+    chartOptions.plugins.title.text =
```

Comment:
what if `chartOptions` is undefined.

Reply:
ChartOptions is a predefined config. So it will not be undefined